### PR TITLE
Added support for stable tag releases

### PR DESCRIPTION
**Pull request by 0xsouravm**
This PR fixes the issues #19 and #20 by adding support for stable releases tags.

### CODE REVIEW

File: src/tests.rs [hunk 1]
```diff
@@ -43,11 +43,13 @@ mod tests {
         version: &str,
         input_cargo_toml_path: &Path,
     ) -> Result<Option<String>, Box<dyn Error>> {
-        let mut crates_versions = get_version_mapping_with_fallback(crate::DEFAULT_GIT_SERVER, version)
-            .await
-            .unwrap();
+        let mut crates_versions =
+            get_version_mapping_with_fallback(crate::DEFAULT_GIT_SERVER, version)
```
Comment (ggwpez) [inline src/tests.rs:47]: Please no no-op formatting changes, it makes reviewing slower.

Comment (0xsouravm) [reply src/tests.rs:47]: I am sorry about that, I will make sure I don't do it next time onwards.

File: src/versions.rs [hunk 2]
```diff
@@ -86,6 +87,102 @@ pub struct Workspace {
...
+        .header("Accept", "application/vnd.github.v3+json")
+        .send()
+        .await?;
+    
+        let output = if response.status().is_success() {
+            response.text().await?
+        } else {
+            // query the github api using gh command
+            String::from_utf8(
+                std::process::Command::new("gh")
```
Comment (ggwpez) [inline src/versions.rs:151]: Why is it using gh cli here instead of the reqwest lib?

**Review by ggwpez — COMMENTED** [review]
Design looks good, just some comments. Thanks!

**Comment by 0xsouravm** [comment]
> Design looks good, just some comments. Thanks!

Thanks a lot! Should I revert the formatting back to the old one or is it alright to have it? I will make sure it is not touched further.

**Comment by ggwpez** [comment]
> Thanks a lot! Should I revert the formatting back to the old one or is it alright to have it? I will make sure it is not touched further.

Its fine this time - we should probably have a formatting CI here anyway.

**Comment by 0xsouravm** [comment]
> > Thanks a lot! Should I revert the formatting back to the old one or is it alright to have it? I will make sure it is not touched further.
> 
> Its fine this time - we should probably have a formatting CI here anyway.

Oh alright. Sure. Thanks for the review! Is there anything more you would want me to change?

**Comment by 0xsouravm** [comment]
Can this be merged? @ggwpez

**Review by ggwpez — APPROVED** [review]

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/sponsorship-for-sub0-thailand-2024/10093/26

**Pull request merged**